cmd/lakectl/cmd/fs.go

```diff
@@ -175,16 +173,28 @@ func uploadObject(ctx context.Context, client api.ClientWithResponsesInterface,
 		return nil, err
 	}
 	if resp.StatusCode() != http.StatusCreated {
-		apiErrors := []*api.Error{resp.JSON400, resp.JSON401, resp.JSON404, resp.JSONDefault}
-		for _, apiErr := range apiErrors {
-			if apiErr != nil {
-				return nil, fmt.Errorf("%w: %s", ErrRequestFailed, apiErr.Message)
-			}
+		if resp.JSONDefault != nil {
+			return nil, fmt.Errorf("%w: %s", ErrRequestFailed, resp.JSONDefault.Message)
 		}
+		return nil, fmt.Errorf("%w: %s", ErrRequestFailed, http.StatusText(resp.StatusCode()))
 	}
 	return resp.JSON201, nil
 }
 
+// readSize returns the size of r.
+func readSize(r io.Seeker) (int64, error) {
```

Comment:
I think root.go _also_ performs this very thing, and on trunk I've already refactored (https://github.com/treeverse/lakeFS/blob/933ec252c1ed0176581d9887db615f4abf3354e8/pkg/api/client.go#L746).  Let's add a TODO(ariels) to move it to helpers?